Userland/Libraries/LibWasm/AbstractMachine/Validator.cpp

```diff
@@ -384,176 +388,176 @@ VALIDATE_INSTRUCTION(f64_const)
 // https://webassembly.github.io/spec/core/bikeshed/#-tmathsfhrefsyntax-unopmathitunop
 VALIDATE_INSTRUCTION(i32_clz)
 {
-    if (stack.is_empty() || !stack.last().is_of_kind(ValueType::I32))
-        return Errors::invalid_stack_state(stack, Tuple { ValueType(ValueType::I32) });
+    TRY(stack.take<ValueType::I32>());
+    stack.append(ValueType(ValueType::I32));
     return {};
 }
 
 VALIDATE_INSTRUCTION(i32_ctz)
 {
-    if (stack.is_empty() || !stack.last().is_of_kind(ValueType::I32))
-        return Errors::invalid_stack_state(stack, Tuple { ValueType(ValueType::I32) });
+    TRY(stack.take<ValueType::I32>());
+    stack.append(ValueType(ValueType::I32));
```

Comment:
This can be written as
```suggestion
    TRY(stack.take_and_put<ValueType::I32>(ValueType::I32));
```

Similarly for most of the `T -> T` ops.

A slight optimisation can then be introduced (later, if needed) to avoid vector push/pop churn.